### PR TITLE
refactor: Improved performance by using async functions with promises

### PR DESCRIPTION
**Pull request by rohitp-bsf**
### Description
- As we need to register settings and controls initially before using, it's usually taking a time to loop over thousands of array of the objects which blocking a customizer page to load.
- Made function asynchronous and wait for the promise to complete which will not wait for the entire loop and dependent function will execute after the async execution.

### Screenshots
NA

### Types of changes
- Made possible functions asynchronous.
- Added dependent functions inside promises. 

### How has this been tested?
- Check the performance benefit by comparing the previous version.
- Check possible actions like partial refresh, tab switch, grid drop, etc.

### Checklist:
- [ ] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

**Review by vrundakansara — APPROVED** [review]

File: inc/assets/js/customizer-preview.js [hunk 1]
```diff
@@ -38,6 +38,22 @@
             }
         );
 
+		/**
+		 * Register partial refresh events at once asynchronously.
+		 */
+		wp.customize.preview.bind( 'active', function() {
+			var partials = $.extend({}, astraCustomizer.dynamic_partial_options);
+			Object.keys(partials).forEach(async function ( key) {
```
Comment (premanshup) [inline inc/assets/js/customizer-preview.js:46]: @rohitp-bsf This is ES6 nad may not work in the browsers where ES6 is not supported. Here is a polyfill which can be used for this - 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys#polyfill

File: package.json [hunk 2]
```diff
@@ -32,9 +32,10 @@
     "kind-of": ">=6.0.3",
     "minimist": ">=1.2.3",
     "node-sass": "^4.14.1",
-    "postcss-flexibility": "^2.0.0"
+    "postcss-flexibility": "^2.0.0",
+    "@wordpress/scripts": "^12.6.1"
```
Comment (premanshup) [inline package.json:36]: @rohitp-bsf Can you please explain why we are including this in our main package.json file?

File: inc/assets/js/customizer-preview.js (hunk 1 above)
Comment (rohitp-bsf) [reply inc/assets/js/customizer-preview.js]: Resolved! @premanshup

File: package.json (hunk 2 above)
Comment (rohitp-bsf) [reply package.json]: Removed! Thanks @premanshup

**Review by premanshup — APPROVED** [review]

**Pull request merged**